platform-sdk/docs/proposals/001-merkle-state-api/001-merkle-state-api.md

```diff
@@ -0,0 +1,74 @@
+
+## Requirements
+
+- `PlatformState` should be refactored to a singleton object as defined by the State API.
+- the Merkle tree should be reshaped so that `MerkleHederaState` becomes the root of the tree and `PlatformState` becomes its child.
+- there should be a new module - `swirlds-state-api` - to host a set of interfaces, records, and abstract classes that represent the Hedera state.
+  This module should have a minimal set of dependencies. Ideally, the Block Node should not have a compile-time dependency
+  on any other modules but this one to interact with the state.
+- there should be a new module - `swirlds-state-impl` module to host the implementation of these interfaces and helper classes.
+
+## Changes
+
+### Public API
+
+State API classes and interfaces will migrate to a designated module. The classes themselves will not change.
+
+### Modules and Repos
+
+Two new modules are added as a part of this proposal:
+
+![state-modules](assets/state-modules.png)
+
+Note that the `swirlds-state-api` is already created and has a set of classes by [this PR](https://github.com/hashgraph/hedera-services/pull/12570). There is also one [more PR](https://github.com/hashgraph/hedera-services/pull/13331) related to this change.
+
+There is one more change that can be relevant in the context of public APIs. This is the proposed change to the Merkle tree structure.
+
+![new-merkle-tree-structure.png](assets/new-merkle-tree-structure.png)
+
+## Components and Architecture
+
+### Platform State
+
+Currently `PlatformState` is a special case node of the merkle tree. As a part of the simplification effort **it should be refactored to a singleton state**.
+
+### KV Node
+
+We need to have a class that represents key-value node in the tree, as we have it for singleton and queue nodes. Currently, this class is missing. 
+
+### Merkle state refactoring
+
+- Move `PlatformState` from the root to `MerkleHederaState`
+- Replace usage of existing root of `State` type with `MerkleHederaState`
+  - It would be much easier to do, if they implemented the same interface (e.g. `RootNode`). Currently, there is now direct relation between these classes.
+- Make sure that the state migration happens properly during the version update
+
+## Test plan
+
+This design doesn't add any new functionality. Therefore, the assumption is that as long as all existing tests are passing (unit tests, integration tests, and performance tests), the change is considered to be tested.
+
+## Implementation and delivery plan
+
+- Move State API implementation classes from `swirlds-platform-core` to `swirlds-state-impl`.
+- Add common interface for `MerkleHederaState` and `State` to make the migration easier.
+- Make `MerkleHederaState` a root of the Merkle tree.
+- Implement a class that represents KV node.
+- Refactor `PlatformState` into a singleton. This depends on [migrating PlatformState to protobuf](https://github.com/hashgraph/hedera-services/pull/13694). Migrate the data.
```

Comment:
The PR link seems to be wrong. From the text, I expected a PR that defines the `PlatformState` in protobuf.

Also, the format of `PlatformState` as protobuf is a public API. I think this information needs to be included as part of this proposal or at least linked and explained how it relates to this one since it is a dependency.

Reply:
I removed the PR link and added proto definition to the proposal